UKHomeOffice/nmsw-ui
# NMSW-1307 Update task details page to use /declaration endpoint

# Ticket



----

## AC

We're moving to
`/declaration/<id>` for declaration data
`/declaration/<id>/attachments` for when you need the attachments

FE needs updating to match this

----

## To test
Draft report

- create a report
- upload a fal 1
- go to task page
- > FAL5 shows as required
- upload a FAL5
- > task page should show FAL5 as completed
- > FAL 6 should show as required
- click on passengers
- select 'no'
- return to task page
- > passengers should show as complete
- > CYA link should be enabled
- click on passengers
- select 'yes'
- return to task page
- > passengers should show as in progress
- > CYA link should be disabled
- upload a FAL 6
- return to task page
- > passengers should show as in progress
- click on passengers
- click on final file
- return to task page
- > passengers should show as complete
- > CYA link should be enabled
- click delete draft
- delete the draft
- > you should be returned to the dashboard with a confirmation banner
- try to return to that declaration task page by typing url in browser
- > you should receive a something went wrong message
- click on click here to continue
- > you should be taken to the dashboard 


----

## Notes
Updated the logic for determining 'complete' status to use the counts from declaration rather than just checking a file exists as this is more accurate. This enables us to wait for amg processing to be completed before we change state.
